File: 문제풀이/온라인 저지/프로그래머스/# 02. Level 2/# 131701. 연속 부분 수열 합의 개수/kotlin/Solution.kt
class Solution {
    fun solution(elements: IntArray): Int {
        var rt: Int
        var sum: Int
        val set = HashSet<Int>()
        val size = elements.size
        for (lt in 0 until size) {
            sum = 0
            rt = lt
            do {
                sum += elements[rt]
                set += sum
                rt = (rt + 1 + size)%size
            } while (rt != lt)
        }
        return set.size
    }
}

File: 문제풀이/온라인 저지/프로그래머스/# 02. Level 2/# 131701. 연속 부분 수열 합의 개수/python/solution.py
def solution(elements):
    sum_set = set()
    for lt in range(len(elements)):
        rt = lt
        sum = 0
        while True:
            sum += elements[rt]
            sum_set.add(sum)

            rt = (rt + 1)%len(elements)
            if lt == rt:
                break
    return len(sum_set)
